corpus/segments.py

```diff
@@ -172,13 +175,16 @@ def __init__(self, segment_file, split=None, shuffle=True, shuffle_seed=0x3C5EA3
         self.shuffle = shuffle
         self.shuffle_seed = shuffle_seed
 
-        self.out_segments = {k: self.output_path("%s.segments" % k) for k in self.split.keys()}
+        self.out_segments = {
+            k: self.output_path("%s.segments" % k if not zip_output else "%s.segments.gz" % k)
```

Comment:
```suggestion
            k: self.output_path(f"{k}.segments{'.gz' if zip_output else ''}")
```
shorter + use f-string ?